### PR TITLE
APP-611 origin check multiple ips

### PR DESCRIPTION
**Pull request by ecarrenho-daitan**
* Build utility class to load log message resource from resource bundles.
* Exceptions using varargs list of strings to better support a single solution entry (had to instantiate a List collection, now it is just a matter of passing a single string as argument).

### CODE REVIEW

File: integration-commons/src/main/java/org/symphonyoss/integration/exception/ExceptionMessageFormatter.java [hunk 1]
```diff
@@ -79,12 +77,31 @@ public static String format(String component, String message,
...
+   * @param component The component where the exception was thrown.
+   * @param message The message why the exceptions happened.
+   * @param solution The solution provided to the exceptions.
+   * @return Formatted message
+   */
+  public static String format(String component, String message,
+      String solution) {
+    List<String> solutions = new ArrayList<>();
+    solutions.add(solution);
+    return getMessage(component, message, solutions, null);
```
Comment (robson-sanchez) [inline integration-commons/src/main/java/org/symphonyoss/integration/exception/ExceptionMessageFormatter.java]: Could we use varargs parameters insteadof a list of strings?

Something like that: getMessage(String component, String message, Throwable t, String... solutions)

**Review by robson-sanchez — APPROVED** [review]
Minor changes to simplify the code.

File: integration-commons/src/main/java/org/symphonyoss/integration/exception/ExceptionMessageFormatter.java (hunk 1 above)
Comment (ecarrenho-daitan) [reply integration-commons/src/main/java/org/symphonyoss/integration/exception/ExceptionMessageFormatter.java]: Done

File: integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ * Created by Evandro Carrenho on 30/01/17.
+ */
+@Component
+public class LogMessageSource {
+
+  private static final ISymphonyLogger LOGGER = SymphonyLoggerFactory.getLogger(LogMessageSource.class);
+
+  private static final String UNABLE_TO_LOAD_LOG_MESSAGES = "integration.commons.unable.to.load.log.messages";
+
+  private static final String UNABLE_TO_LOAD_LOG_MESSAGES_SOLUTION1 =
```
Comment (dnsymphony) [inline integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java]: What's the '1' for?

File: integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+   * Retrieves the indicated resource and substitutes the placeholders with the given string arguments.
+   * @param resId The resource id to be retrieved.
+   * @param args The strings to fill in placeholders.
+   * @return Formatted string.
+   */
+  public String getMessage(String resId, String... args) {
+    try {
+      return accessor.getMessage(resId, args);
+    } catch (NoSuchMessageException e) {
+      return UNABLE_TO_RECORD_LOG_INFORMATION_LOG_MESSAGE_NOT_FOUND;
```
Comment (dnsymphony) [inline integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java]: This should include the resource ID and the args.

File: integration-commons/src/test/java/org/symphonyoss/integration/logging/LogMessageSourceTest.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+@ContextConfiguration(classes = {LogMessageSource.class})
+public class LogMessageSourceTest {
+
+  @Autowired
+  LogMessageSource logMessage;
+
+  @Test
+  public void testCommonsMessage() {
+
+    assertEquals("Unable to load log messages from log-messages.properties files. "
```
Comment (dnsymphony) [inline integration-commons/src/test/java/org/symphonyoss/integration/logging/LogMessageSourceTest.java:30]: This is testing the text in the resource bundle in the test directory (which right now is a copy of the one in src directory.  But if someone changes that one, this test will still pass.  Is that expected?

**Review by dnsymphony — CHANGES_REQUESTED** [review]

File: integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java (hunk 2 above)
Comment (ecarrenho-daitan) [reply integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java]: Removed. I was numbering solutions as they might have been multiple, but it ended up being weird.

File: integration-commons/src/test/java/org/symphonyoss/integration/logging/LogMessageSourceTest.java (hunk 4 above)
Comment (ecarrenho-daitan) [reply integration-commons/src/test/java/org/symphonyoss/integration/logging/LogMessageSourceTest.java:30]: Yes, it is expected. We are testing the code that accesses resources not the resources themselves.

File: integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java (hunk 3 above)
Comment (ecarrenho-daitan) [reply integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java]: Done.

**Pull request merged**